### PR TITLE
Juan SGW 625 unify evm providers

### PR DESCRIPTION
**Pull request by juanolmedo1**
Ticket: [SGW-625](https://constellationnetwork.atlassian.net/browse/SGW-625)

[SGW-625]: https://constellationnetwork.atlassian.net/browse/SGW-625?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: source/scripts/common/availableChainMethods.ts [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
 import { AvailableMethods } from './availableMethods';
 import { AvailableEthereumChainMethods } from './methods/availableEthereumMethods';
-import { AvailablePolygonChainMethods } from './methods/availablePolygonMethods';
-import { AvailableBSCChainMethods } from './methods/availableBSCMethods';
-import { AvailableAvalancheChainMethods } from './methods/availableAvalancheMethods';
 import { AvailableConstellationChainMethods } from './methods/availableConstellationMethods';
 
 type AvailableChainMethod = {
-  chain: StargazerChain;
+  chain: StargazerChain | StargazerProvider;
```
Comment (juandavidkincaid) [inline source/scripts/common/availableChainMethods.ts:7]: As well keeping an union here seems odd to me for some reason.

File: source/scripts/common/chains.ts [hunk 2]
```diff
@@ -1,5 +1,10 @@
 import { AssetType } from "state/vault/types";
 
+enum StargazerProvider {
+  CONSTELLATION = 'constellation',
+  ETHEREUM = 'ethereum',
+}
+
 enum StargazerChain {
   CONSTELLATION = 'constellation',
   ETHEREUM = 'ethereum',
```
Comment (juandavidkincaid) [inline source/scripts/common/chains.ts:10]: I just followed the references of `StargazerChain` and `StargazerProvider`, I may think you need to cleanup/reorganize uses for both as I think you're mixing things up with those.

File: source/scripts/Background/dappRegistry/dappRegistry.ts [hunk 3]
```diff
@@ -61,7 +61,7 @@ class DappRegistry {
 
   sendOriginChainEvent(
     origin: '*' | string,
-    chain: '*' | StargazerChain,
+    chain: '*' | StargazerChain | StargazerProvider,
```
Comment (juandavidkincaid) [inline source/scripts/Background/dappRegistry/dappRegistry.ts:64]: Don't like this unions.

File: source/scripts/Background/controllers/WalletController.ts [hunk 4]
```diff
@@ -358,59 +360,68 @@ class WalletController implements IWalletController {
...
     }
 
     if (network === KeyringNetwork.Ethereum) {
       this.account.networkController.switchEthereumChain(chainId as EthChainId);
+      masterController.ethereumProvider.switchProvider(StargazerChain.ETHEREUM);
       if (!isNative) {
+        const hexChainId = `0x${ETH_NETWORK[chainId].chainId.toString(16)}`;
         getDappRegistry().sendOriginChainEvent(
           '*',
           StargazerChain.ETHEREUM,
```
Comment (juandavidkincaid) [inline source/scripts/Background/controllers/WalletController.ts:380]: Yeap I think this is mixed. (`StargazerProvider` vs `StargazerChain`)

File: source/scenes/settings/Networks/Networks.container.tsx [hunk 5]
```diff
@@ -17,9 +17,9 @@ const NetworksContainer: FC = () => {
   const { activeNetwork, customNetworks }: IVaultState = useSelector((state: RootState) => state.vault);
   const linkTo = useLinkTo();
 
-  const handleChangeNetwork = (networkType: string, networkId: string) => {
+  const handleChangeNetwork = async (networkType: string, networkId: string) => {
     if (activeNetwork[networkType as keyof ActiveNetwork] !== networkId) {
-      walletController.switchNetwork(networkType, networkId);
+      await walletController.switchActiveNetwork(networkId);
```
Comment (juandavidkincaid) [inline source/scenes/settings/Networks/Networks.container.tsx:22]: It seems this `await` is not necessary.

**Review by juandavidkincaid — CHANGES_REQUESTED** [review]
I think changes, in general, are in accordance with the ticket (solve it), but unfortunately, this time it is really confusing to navigate through the codebase, I know the codebase per se is messy, but I would highly suggest cleanup things while you add features, or try to.

And definitely please address the uses of `StargazerChain` and `StargazerProvider(enum)`.

File: source/scripts/Provider/EVMProvider.ts [hunk 6]
```diff
@@ -353,7 +355,7 @@ export abstract class EVMProvider implements IRpcChainRequestHandler {
         walletId: activeWallet.id,
         walletLabel: activeWallet.label,
         publicKey: '',
-        chain: this.getNetworkId(),
+        chain: StargazerChain.ETHEREUM,
         chainLabel
```
[the remark this answers is not in the document]
Comment (juanolmedo1) [reply source/scripts/Provider/EVMProvider.ts:359]: Yeah, it's intended. The chain value is used to select the expected provider: Constellation or Ethereum.
I'll update the property name to `provider`

**Review by juandavidkincaid — APPROVED** [review]
Looks nice :]]]], let me know when it is deployed on chrome.

**Pull request merged**